packages/aws-cdk/lib/cdk-toolkit.ts

```diff
@@ -127,6 +128,14 @@ export class CdkToolkit {
       return this.watch(options);
     }
 
+    if (options.notificationArns) {
+      options.notificationArns.map( arn => {
+        if (!validateSnsTopicArn(arn)) {
```

Comment:
```suggestion
        if (!Token.unresolved(arn) && !validateSnsTopicArn(arn)) {
```

Also please add a test for this condition.

Reply:
@otaviomacedo I have looked into it and can indeed add the !Token.isUnresolved(arn) condition to the check of notification arn. However, as this check happens prior to stack synthesis, it seems redundant to check if the token has been resolved. 
Let me know if I am misunderstanding :)

Reply:
`arn` may be a "normal" string (in which case it makes sense to validate it) or it may a token, which is a special kind of string, that is a stand-in for something else (in which case it doesn't make sense to validate it because it will always fail). In the second case, `Token isUnresolved()` returns true.

Reply:
@otaviomacedo thanks for the reply! I do get that tokens can be passed around and are converted to strings during synthesis to CloudFormation templates.
What I am thinking is that the Arn in this specific case is passed as a parameter when using the CLI. for example:
`cdk deploy --all --notification-arns arn:aws:sns:us-east-1::some-sns-topic`
in which case the arn will never be a token as these only exist in a stack prior to synthesis.

Again, I might be wrong, but is there a case where the arn is passed to the CLI command as a token?

Reply:
@otaviomacedo Readng around I found #8581 which does ask to implement a way to add the notification arn to the stack prior to synthesis. The issue has been open for a while, but would require the Token.isUnresolved() in the check.

What do you think? should we add it just in case?